src/trace-plugin-loader.js

```diff
@@ -116,16 +116,20 @@ function activate(agent) {
       if (!patchSet) {
         // Load the plugin object
         var plugin = originalModuleLoad(instrumentation.file, module, false);
-        patchSet = [];
+        patchSet = {};
```

Comment:
I don't think the original hook loading mechanism had anything stored as an array. I think I introduced `patchSet` as an array in the plugin loader and changed my mind about it later.